src/eventProcessor.ts

```diff
@@ -336,52 +343,94 @@ export default async (event: ContractEvent): Promise<void> => {
 
       verbose('Extension with address:', extensionAddress, 'was enabled');
 
-      await mutate('updateColonyExtensionByAddress', {
+      const {
+        data: {
+          updateColonyExtension: { extensionHash, colonyAddress },
+        },
+      } = await mutate('updateColonyExtensionByAddress', {
         input: {
           id: extensionAddress,
           isInitialized: true,
         },
       });
 
+      /* Listen for motions once Voting Reputation is enabled.*/
+      if (getExtensionHash(Extension.VotingReputation) === extensionHash) {
+        await addMotionEventListener(
+          ContractEventsSignatures.MotionCreated,
+          colonyAddress,
+        );
+      }
       return;
     }
 
-    case ContractEventsSignatures.TokensMinted: {
+    /* Entry point for a newly created motion. */
+    case ContractEventsSignatures.MotionCreated: {
       const {
-        transactionHash,
-        blockNumber,
         contractAddress: colonyAddress,
+        args: { motionId },
       } = event;
-      const {
-        agent: initiatorAddress,
-        who: recipientAddress,
-        amount,
-      } = event.args;
 
-      const tokenAddress = await getColonyTokenAddress(colonyAddress);
-
-      verbose(
-        amount.toString(),
-        'tokens were minted in colony:',
-        colonyAddress,
+      const colonyClient = await networkClient.getColonyClient(colonyAddress);
+      const parsedAction = await getParsedActionFromMotion(
+        motionId,
+        colonyClient,
       );
 
-      await mutate('createColonyAction', {
-        input: {
-          id: transactionHash,
-          colonyId: colonyAddress,
-          type: ColonyActionType.MintTokens,
-          initiatorAddress,
-          recipientAddress,
-          amount: amount.toString(),
-          blockNumber,
-          tokenAddress,
-        },
-      });
+      if (parsedAction) {
+        const contractOperation = parsedAction.name;
+
+        /* Handle the action type-specific mutation here*/
+        switch (contractOperation) {
+          case ColonyOperations.MintTokens: {
+            await writeMintTokensMotionToDB(event, parsedAction, colonyClient);
+            break;
+          }
+          default: {
+            break;
+          }
+        }
 
-      return;
+        verbose(`${contractOperation} Motion Created`);
+      }
     }
 
+    // case ContractEventsSignatures.TokensMinted: {
```

Comment:
Only commented out for testing motions. I'm noticing the TokensMinted event is firing after the CreateMotion event. A look at redux tells us the correct saga is being called (rootMotion), so not sure why this is happening.